app/controllers/api/groups_controller.rb

```diff
@@ -305,6 +305,54 @@ def remove_tag
       render 'shared/http_status', locals: { code: '404', message: I18n.t('tags.not_found') }, status: :not_found
     end
 
+    def extension
+      grouping = Grouping.find_by(group_id: params[:id], assignment: params[:assignment_id])
+      case request.method
+      when 'DELETE'
+        if grouping.extension.present?
+          grouping.extension.destroy!
+          # Successfully deleted the extension; render success
+          render 'shared/http_status', locals: { code: '200', message:
+            HttpStatusHelper::ERROR_CODE['message']['200'] }, status: :ok
+        else
+          # cannot delete a non existent extension; render failure
+          render 'shared/http_status', locals: { code: '422', message:
+            HttpStatusHelper::ERROR_CODE['message']['422'] }, status: :unprocessable_entity
+          nil
+        end
+      when 'POST'
+        extension_values = extension_params
```

Comment:
Ah, you can move this into the `if` branch (same in the `PATCH` branch)

Reply:
I'm not what you mean here, could you please clarify? thank you.

Reply:
Move this line and the next into the if `grouping.extension.nil?` branch below, because these variables are only being used in that branch (and not the else branch)

Reply:
oh i see, yes good point. Thanks.